plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend

```diff
@@ -378,6 +379,40 @@ class ASTStructureValidator {
 			constraints
 		)
 	}
+	
+	def private dispatch void validateASTStructure(
+		CoalesceExpression model,
+		ASTStructureDiagnosticProducer producer,
+		Set<LabelledStatement> validLabels,
+		Constraints constraints
+	) {
+		val container = model.eContainer 
+		if (container instanceof BinaryLogicalExpression) {
+			val target = NodeModelUtils.findActualNodeFor(model)
+			producer.node = target
+			producer.addDiagnostic(
+					new DiagnosticMessage(IssueCodes.getMessageForAST_INVALID_COALESCE_PARENT(container.op.literal),
+						IssueCodes.getDefaultSeverity(IssueCodes.AST_INVALID_COALESCE_PARENT), IssueCodes.AST_INVALID_COALESCE_PARENT))
+		} else if (model.expression instanceof BinaryLogicalExpression) {
+			val target = NodeModelUtils.findActualNodeFor(model.expression)
+			producer.node = target
+			producer.addDiagnostic(
+					new DiagnosticMessage(IssueCodes.getMessageForAST_INVALID_COALESCE_CHILD((model.expression as BinaryLogicalExpression).op.literal),
+						IssueCodes.getDefaultSeverity(IssueCodes.AST_INVALID_COALESCE_CHILD), IssueCodes.AST_INVALID_COALESCE_CHILD))
+		} else if (model.defaultExpression instanceof BinaryLogicalExpression) {
```

Comment:
I think an example for this case would be this: `a ?? (b && c)`? (I couldn't find a test case.)

Reply:
Done